xdsl/ir.py

```diff
@@ -659,6 +659,30 @@ def add_region(self, region: Region) -> None:
         self.regions.append(region)
         region.parent = self
 
+    def get_region_index(self, region: Region) -> int:
+        """Get the region position in the operation."""
+        if region.parent is not self:
+            raise Exception("Region is not inside the operation.")
+        for idx, curr_region in enumerate(self.regions):
+            if curr_region is region:
+                return idx
+        assert False, "Unexpected xdsl error"
```

Comment:
I don't like this assert. I think you could raise another Exception here?
Though, indeed the IR is somehow corrupted if we reach that point (self is region's parent, but parent doesn't contain self). Still that could be explained through an Exception I believe.

Reply:
I think there is an exception that we can use that show an incorrect state in the IR.
However, I'm also fine with an assert, as these kind of things should not happen unless our code is wrong (or the user does shady things).
I would however write a small description.